Predefined colors

tbaranes opened the pull request:
Issue #71 

I wasn't sure where `predefinedColor` should be. I finally put it in `FillDesignable` since it seems to be the best place. It works well, and as `predefinedGradient`, it will set its color only if `fillColor` isn't set.

Commit:
Fix gradient rotation...

Commit:
Update CHANGELOG

Commit:
Merge branch 'master' into hotfix/designable_gradient_rotation

Commit:
Update CHANGELOG

Commit:
Script to generate predefined colors

Commit:
Script to generate predefined colors

Commit:
New IB property: predefinedColor

Commit:
New IB property: predefinedColor

Commit:
Update docs

Commit:
Update docs

JakeLin committed:
Merge pull request #91 from JakeLin/hotfix/designable_gradient_rotation

Fixed gradient designable rotation

JakeLin commented:
Very good, later on, we can add on Material Colorado when we work on v4 material design elements.

JakeLin commented on IBAnimatable/FillDesignable.swift:
```diff
@@ -34,6 +37,62 @@ public extension FillDesignable where Self: UITableViewCell {
+    }
+    
+    switch colorType {
+    case .Emerland:
```
Do you think we should prefix them as FlatEmerland to support MaterialRed later on ?

tbaranes replied:
I don't think so, at least, not until we have to. These flat colors are known named, we risk to confuse their usage if we change the name.

JakeLin replied:
@tbaranes Sorry, I am not sure are we on the same page? I mean for all flat colors, can we add a prefix for all of them like `FlatEmerland` and `FlatPomegranate` instead of `Emerland` and `Pomegranate`. 

The advantage of add `Flat` as prefix, the user knows they are from flat colors (https://gist.githubusercontent.com/tkrugg/a577c32e93eecc6c7991/raw/f7866132e26f9bffcda9caf13dced55e4a99e145/flatuicolors.json). And when we add other set of colors, e.g. Material Design color (https://www.google.com/design/spec/style/color.html#color-color-palette), then all of them will prefix as `Material`, e.g.  `MaterialRed500` and  `MaterialRed50`, those names may not be the final one, just an example to describe the usage. Another benefit is when the user uses the enum in code, when they can start typing `.Flat` and Xcode will auto complete the rest for them.

But there are disadvantage like getting the name longer than it should be.

Please let me know what you think.

tbaranes replied:
I understand your point. I like the idea to have the completion when tapping .Flat. I just updated the PR. For the name, it's predefined colors, so I don't think it's a big issue to have longer name.

lexrus committed:
Fix #95.

JakeLin committed:
Update CHANGELOG.md to reflect the fix for #95

Commit:
Prefix all predefined colors by Flat

Commit:
Prefix all predefined colors by Flat

tbaranes commented:
I like the idea to have the completion when tapping .Flat. I just updated the PR to add the prefix "Flat" for each color. The name is a bit longer, but I don't think it's a big deal since it's just _Predefined flat colors_

Commit:
Fix FillDesignable implementation in AnimatableTableView and AnimatableTableViewCell

commit 2a19aa0b2ee5647d8c9a443ba8ae6fd8139ba374
Author: Tom Baranes <tom.baranes@gmail.com>
Date:   Sun Feb 14 09:53:48 2016 +0100

Commit:
Fix FillDesignable implementation in AnimatableTableView and AnimatableTableViewCell

commit 2a19aa0b2ee5647d8c9a443ba8ae6fd8139ba374
Author: Tom Baranes <tom.baranes@gmail.com>
Date:   Sun Feb 14 09:53:48 2016 +0100

tbaranes commented:
Since, the issue #94  is related to this one, I updated the implementation of all the `Animatable...` `FillDesignable` in order to have only one way to conform the protocol.

JakeLin committed:
Update IBAnimatable.podspec for release tag 1.1

JakeLin committed:
Remove unnecessary `configFillColor` calls

JakeLin committed:
Merge branch 'features/flat_colors' of https://github.com/JakeLin/IBAnimatable into features/flat_colors

JakeLin commented:
Well done @tbaranes , just merged it into master.
